File: shield-spread.js
// const figures = {
//     Via:[1,1,1,1],
//     Cauda Draconis:[1,1,1,0],
//     Puer:[1,1,0,1],
//     Fortuna Minor:[1,1,0,0],
//     Puella:[1,0,1,1],
//     Amissio:[1,0,1,0],
//     Carcer:[1,0,0,1],
//     Laetitia:[1,0,0,0],
//     Caput Draconis:[0,1,1,1],
//     Conjunctio:[0,1,1,0],
//     Acquisitio:[1,1,1,1],
//     Rubeus:[0,1,0,0],
//     Fortuna Major:[0,0,1,1],
//     Albus:[0,0,1,0],
//     Tristitia:[0,0,0,1],
//     Populus:[0,0,0,0] 
// };

const figures = {
    'Via':[1,1,1,1],
    'Cauda Draconis':[1,1,1,0],
    'Puer':[1,1,0,1],
    'Fortuna Minor':[1,1,0,0],
    'Puella':[1,0,1,1],
    'Amissio':[1,0,1,0],
    'Carcer':[1,0,0,1],
    'Laetitia':[1,0,0,0],
    'Caput Draconis':[0,1,1,1],
    'Conjunctio':[0,1,1,0],
    'Acquisitio':[1,1,1,1],
    'Rubeus':[0,1,0,0],
    'Fortuna Major':[0,0,1,1],
    'Albus':[0,0,1,0],
    'Tristitia':[0,0,0,1],
    'Populus':[0,0,0,0]
};

const check = (num) => { 
    if (num==1) {
        return 1
    } else if (num%2<1) {
        return 0
    } else {
        return 1
    }
};

const mixFigs = (fig1, fig2) => {
    let newFig = []
    for (i = 0; i<4; i++) {
        newFig.push(check(fig1[i]+fig2[i]))
    };
    return newFig
};

const genDaughter = (mothers_list, ind) => {
    let daughter = [];
    for (i = 0; i<4; i++) {
        daughter.push(mothers_list[i][ind])
    };
    return daughter
};

var m1 = [1,1,0,0];
var m2 = [0,1,0,1];
var m3 = [0,0,0,0];
var m4 = [1,0,1,1];
var mothers = [m1,m2,m3,m4];
console.log(mothers[i]);

var d5 = genDaughter(mothers,0);
var d6 = genDaughter(mothers,1);
var d7 = genDaughter(mothers,2);
var d8 = genDaughter(mothers,3);
var daughters = [d5,d6,d7,d8];
console.log(daughters[i]);


var n9 = mixFigs(m1,m2);
var n10 = mixFigs(m3,m4);
var n11 = mixFigs(d5,d6);
var n12 = mixFigs(d7,d8);
var nieces = [n9,n10,n11,n12];
console.log(nieces);

var w13 = mixFigs(n9,n10);
var w14 = mixFigs(n11,n12);
var witnesses = [w13,w14]
console.log(witnesses);

var j15 = mixFigs(w13,w14);
console.log(j15);

var s16 = mixFigs(j15,m1);
console.log(s16);

File: shield-spread.py
#Generates a geomancy shield spread chart given an input of the 4 Mothers either as a list of 4 numbers (0 or 1) or by input name (eg. Fortuna Minor)

def check(num):
    if num==1:
        return 1
    elif num%2 < 1:
        return 0
    else:
        return 1

def gen(fig1,fig2):
    return [check(fig1[i]+fig2[i]) for i in range(4)]

figures = {
    'Via':[1,1,1,1],
    'Cauda Draconis':[1,1,1,0],
    'Puer':[1,1,0,1],
    'Fortuna Minor':[1,1,0,0],
    'Puella':[1,0,1,1],
    'Amissio':[1,0,1,0],
    'Carcer':[1,0,0,1],
    'Laetitia':[1,0,0,0],
    'Caput Draconis':[0,1,1,1],
    'Conjunctio':[0,1,1,0],
    'Acquisitio':[1,1,1,1],
    'Rubeus':[0,1,0,0],
    'Fortuna Major':[0,0,1,1],
    'Albus':[0,0,1,0],
    'Tristitia':[0,0,0,1],
    'Populus':[0,0,0,0]
}

m1,m2,m3,m4 = [],[],[],[]
mothers = [m1,m2,m3,m4]
print(mothers)

#Daughters - calculated by position of mothers. m1 = fire, m2 = air, m3 = water, m4 = earth.
#Each daughter is calculated by position
d5 = [i[0] for i in mothers]
d6 = [i[1] for i in mothers]
d7 = [i[2] for i in mothers]
d8 = [i[3] for i in mothers]
daughters = [d5,d6,d7,d8]
print(daughters)

#Nieces
n9 = gen(m1,m2)
n10 = gen(m3,m4)
n11 = gen(d5,d6)
n12 = gen(d7,d8)
nieces = [n9,n10,n11,n12]
print(nieces)

#Witnesses
w13 = gen(n9,n10)
w14 = gen(n11,n12)
witnesses = [w13,w14]
print(witnesses)

#Judge and Sentence
j15 = gen(w13,w14)
s16 = gen(j15,m1)

print(j15)
print(s16)